logging/logrus/middleware.go

```diff
@@ -59,14 +60,22 @@ func newServerRequestFields(req *http.Request) logrus.Fields {
 	if host == "" {
 		host = req.Host
 	}
+
 	fields := logrus.Fields{
 		"system":                    SystemField,
 		"span.kind":                 "server",
 		"http.url.path":             req.URL.Path,
 		"http.proto_major":          req.ProtoMajor,
-		"http.request.length_bytes": req.ContentLength,
 		"http.host":                 host,
+		"http.request.method":       req.Method,
+		"http.request.user_agent":   req.Header.Get("User-Agent"),
+		"http.request.length_bytes": req.ContentLength,
 	}
+
+	if req.Referer() != "" {
```

Comment:
Why is referer a special case?

Reply:
`referer` is not always present on the request so would rather not add it to the logs if it is not present. Can be convinced otherwise :)

Reply:
I thought so, I am fine with that.

Reply:
I would prefer to keep the empty field in as this will make sure that all log lines will have the same fields.

Reply:
@yifanzz @jutkko have updated so `http.request.referer` field is always present even if `""`.